### PR TITLE
Split exchange.py module into several dedicated modules

### PR DESCRIPTION
**Pull request by fbeutin-ledger**

### CODE REVIEW

File: test/python/test_fake_signer.py [hunk 1]
```diff
@@ -89,70 +86,66 @@ def test_fake_partner_credentials_signed(self, backend, operation):
...
 
-        ex.init_transaction()
+        transaction_id = ex.init_transaction().data
         ex.set_partner_key(partner.credentials)
         ex.check_partner_key(LEDGER_SIGNER.sign(partner.credentials))
-        ex.process_transaction(TX_INFOS[operation], FEES)
+        tx = craft_tx(operation, TX_INFOS[operation], transaction_id)
+        ex.process_transaction(tx, FEES)
 
+        encoded_tx = encode_tx(operation, partner_fake, tx)
```
Comment (lpascal-ledger) [inline test/python/test_fake_signer.py:101]: Same as before, this manual encoding should be invisible for the user.

**Review by lpascal-ledger — APPROVED** [review]
I think we could improve further by hidding some complexity into the `ExchangeClient` class.

File: test/python/apps/exchange_transaction_builder.py [hunk 2]
```diff
@@ -97,3 +102,31 @@ def create_transaction(self, conf: Dict, transaction_id: bytes) -> bytes:
...
+def extract_payout_ticker(subcommand: SubCommand, tx_infos: Dict) -> str:
+    subcommand_specs = SUBCOMMAND_TO_SPECS[subcommand]
+    return tx_infos[subcommand_specs.payout_field]
+
+def extract_refund_ticker(subcommand: SubCommand, tx_infos: Dict) -> Optional[str]:
+    subcommand_specs = SUBCOMMAND_TO_SPECS[subcommand]
+    if subcommand_specs.refund_field:
+        return tx_infos[subcommand_specs.refund_field]
+    else:
+        return None
```
[the remark this answers is not in the document]
Comment (fbeutin-ledger) [reply test/python/apps/exchange_transaction_builder.py:132]: Done !

[the remark this answers is not in the document]
Comment (fbeutin-ledger) [reply test/python/apps/exchange_transaction_builder.py:132]: If we were designing a high level client for exchange to be used in prod I would 100% agree with you but it's a client to actually send wrong stuff so we should not abstract too much

File: test/python/test_fake_signer.py [hunk 3]
```diff
@@ -89,70 +86,66 @@ def test_fake_partner_credentials_signed(self, backend, operation):
...
+        partner = SigningAuthority(curve=get_partner_curve(operation), name="partner")
+        partner_fake = SigningAuthority(curve=get_partner_curve(operation), name="partner_fake")
 
-        ex.init_transaction()
+        transaction_id = ex.init_transaction().data
         ex.set_partner_key(partner.credentials)
         ex.check_partner_key(LEDGER_SIGNER.sign(partner.credentials))
-        ex.process_transaction(TX_INFOS[operation], FEES)
+        tx = craft_tx(operation, TX_INFOS[operation], transaction_id)
+        ex.process_transaction(tx, FEES)
```
[the remark this answers is not in the document]
Comment (fbeutin-ledger) [reply test/python/test_fake_signer.py:99]: This is the way it was done before, and it was very hard to introduce errors / valid edge cases
In your code snippet, it will be very hard to send a wrong subcommand for the expected transaction for example

However! I do agree that there is a LOT of repetitiveness, but I plan solve it by introducing helpers that perform valid tx flows, (and calling them everywhere we don't do an error), much like the `ExchangeTestRunner`

In the end I plan to have something like this :
```
def test_error_at_step_4(exchange_client, helpersomething):
    helpersomething.run_until_step_4()
    exchange_client.step4(false data)
```

[the remark this answers is not in the document]
Comment (fbeutin-ledger) [reply test/python/test_fake_signer.py:99]: So yes there is a lot of redundancy and it's an issue, but putting it in the client is IMO not the good way to abstract it

File: test/python/apps/exchange_transaction_builder.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (lpascal-ledger) [reply test/python/apps/exchange_transaction_builder.py:132]: That's a very valid argument. I was having a legitimate client user PoV, which is merely 10% of the tests coverage.

File: test/python/apps/exchange.py
[the remark this answers is not in the document]
Comment (fbeutin-ledger) [reply test/python/apps/exchange.py:1]: Done
